spyder_kernels/console/tests/test_console_kernel.py

```diff
@@ -370,5 +370,78 @@ def f(x):
         assert content['found']
 
 
+def test_turtle_launch(tmpdir):
+    """Test turtle scripts launching in the same kernel."""
+    # Command to start the kernel
+    cmd = "from spyder_kernels.console import start; start.main()"
+
+    with setup_kernel(cmd) as client:
+        # Remove all variables
+        client.execute("%reset -f")
+        client.get_shell_msg(block=True, timeout=TIMEOUT)
+
+        # Write turtle code to a file
+        code = """
+import turtle
+wn=turtle.Screen()
+wn.bgcolor("lightgreen")
+tess = turtle.Turtle() # Create tess and set some attributes
+tess.color("hotpink")
+tess.pensize(5)
+
+tess.forward(80) # Make tess draw equilateral triangle
+tess.left(120)
+tess.forward(80)
+tess.left(120)
+tess.forward(80)
+tess.left(120) # Complete the triangle
+
+turtle.bye()
+"""
+        p = tmpdir.join("turtle-test.py")
+        p.write(code)
+
+        # Run code
+        client.execute("runfile(r'{}')".format(to_text_string(p)))
+        client.get_shell_msg(block=True, timeout=TIMEOUT)
+
+        # Verify that the `tess` variable is defined
+        client.inspect('tess')
+        msg = client.get_shell_msg(block=True, timeout=TIMEOUT)
+        content = msg['content']
+        assert content['found']
+
+        # Write turtle code to a file
+        code = """
+import turtle
+wn=turtle.Screen()
+wn.bgcolor("lightgreen")
+tess1 = turtle.Turtle() # Create tess and set some attributes
+tess1.color("hotpink")
+tess1.pensize(5)
+
+tess1.forward(80) # Make tess draw equilateral triangle
+tess1.left(120)
+tess1.forward(80)
+tess1.left(120)
+tess1.forward(80)
+tess1.left(120) # Complete the triangle
+
+turtle.bye()
+"""
```

Comment:
Instead of repeating the same code as above but with `tess1` instead of `tess`, please change this to

```python
code = code + "a = 10"
```

Then if we run this code again and we find `a` is defined, it means things went well and `turtle.bye()` didn't generate an Exception.